### ISSUE TITLE
`Packages` file invalid in repo for Debian Buster

### ISSUE THREAD

**Issue by olifre** [Xpra-org/xpra#4069]
**Describe the bug**
Since about 8 PM CET, I observe the following in a vanilla Debian Buster (Debian 10) container:
```
$ wget -q https://xpra.org/repos/buster/xpra.list -O /etc/apt/sources.list.d/xpra.list
$ apt-get update
Hit:1 http://deb.debian.org/debian buster InRelease
Hit:2 http://deb.debian.org/debian-security buster/updates InRelease
Hit:3 http://developer.download.nvidia.com/compute/cuda/repos/debian10/x86_64  InRelease
Hit:4 http://deb.debian.org/debian buster-updates InRelease
Get:5 https://xpra.org/ buster InRelease [1955 B]
Get:6 https://xpra.org/ buster/main amd64 Packages [88.5 kB]
Fetched 90.5 kB in 1s (94.0 kB/s)
Reading package lists...
E: Encountered a section with no Package: header
E: Problem with MergeList /var/lib/apt/lists/xpra.org_dists_buster_main_binary-amd64_Packages.lz4
E: The package lists or status file could not be parsed or opened.
```

In fact, checking the contents of https://xpra.org/dists/buster/main/binary-amd64/Packages , I find some log messages before actual content starts:
```
 * Waiting in queue... 
 * Loading list of packages.... 
The following packages have to be installed:
 dpkg-dev-1.21.21-2.fc39.noarch Debian package development tools
 dpkg-perl-1.21.21-2.fc39.noarch        Dpkg perl modules
 perl-Digest-SHA-1:6.04-501.fc39.x86_64 Perl extension for SHA-1/224/256/384/512
 perl-Digest-SHA1-2.13-40.fc39.x86_64   Digest-SHA1 Perl module
 perl-Tie-4.6-501.fc39.noarch   Base classes for tying variables
 perl-Time-HiRes-4:1.9775-500.fc39.x86_64       High resolution alarm, sleep, gettimeofday, interval timers
 perl-Time-Piece-1.3401-501.fc39.x86_64 Time objects from localtime and gmtime
 perl-TimeDate-1:2.33-11.fc39.noarch    A Perl module for time and date manipulation
 xz-lzma-compat-5.4.4-1.fc39.x86_64     Older LZMA format compatibility binaries

 * Waiting in queue... 
 * Waiting for authentication... 
 * Waiting in queue... 
 * Downloading packages... 
 * Requesting data... 
 * Testing changes... 
 * Installing packages... 
Package: python2-xpra
...
```

**Comment by totaam** [comment]
How odd!

First I thought that this was related to switching packages locally - (because gpg agent forwarding is rubbish and was causing lots of problems, ie: #4001 and many other tickets).
But the script is the same it always was, and it runs:
```
cmd = "dpkg-scanpackages -m %s > %s" % (scan_path, absolute_pkgfile)
```
So there is no way that we append to that package file...
Perhaps `rsync` somehow got mixed up when I uploaded them?
Looks OK now.

**Comment by olifre** [comment]
Thanks, I can confirm that fixes it on my end. Very odd indeed! 
This is a complete shot in the dark, but since `rsync` never failed me: You are not affected by the [recent ext4 corruption bug](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1057843) hitting mainline kernels?

**Comment by totaam** [comment]
> _The commit got merged in 6.5-rc1 so all stable kernels that have 91562895f803 before 6.5 are corrupting data_

A corruption of some sort was my initial guess, but I'm running `6.6.3-200.fc39.x86_64`
And it's much more likely to be PEBKAC.

**Comment by totaam** [comment]
Finally: https://github.com/Xpra-org/xpra/issues/4170#issuecomment-2014260048
